app/src/modules/app/app.c

```diff
@@ -70,43 +59,75 @@ static void init_app_settings(void)
 	}
 }
 
-static void wait_for_cloud_connection(void)
+static void task_wdt_callback(int channel_id, void *user_data)
 {
-	const struct zbus_channel *chan;
+	LOG_ERR("Watchdog expired, Channel: %d, Thread: %s",
+		channel_id, k_thread_name_get((k_tid_t)user_data));
 
-	while (!zbus_sub_wait(&app, &chan, K_FOREVER)) {
-		if (&CLOUD_CHAN == chan) {
-			if (check_cloud_connection()) {
-				return;
-			}
-		}
-	}
+	SEND_FATAL_ERROR();
 }
 
 static void app_task(void)
 {
-	const struct zbus_channel *chan;
+	int err;
+	const struct zbus_channel *chan = NULL;
+	int task_wdt_id;
+	const uint32_t wdt_timeout_ms = (CONFIG_APP_MODULE_WATCDHOG_TIMEOUT_SECONDS) * MSEC_PER_SEC;
+	const k_timeout_t zbus_wait_ms = K_MSEC(wdt_timeout_ms - 100);
```

Comment:
Maybe we could give a little more margin than 100 ms here? The wait can start quite a while after the watchdog has been kicked, so we could get false positives. 

Or.. maybe we actually have to feed it both before and after `zbus_sub_wait()`? I think we do. If so, `zbus_wait_ms` probably needs to have a little more margin anyway, but not a whole lot if the rest of the system behaves